independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -0,0 +1,504 @@
+                    resources.add(res.nextElement());
+                }
+            }
+        }
+        return Collections.enumeration(resources);
+    }
+
+    private ClassLoaderState getState() {
+        ClassLoaderState state = this.state;
+        if (state == null) {
+            synchronized (this) {
+                state = this.state;
+                if (state == null) {
+                    Map<String, List<ClassPathElement>> elementMap = new HashMap<>();
+                    for (ClassPathElement element : elements) {
+                        for (String i : element.getProvidedResources()) {
+                            if (i.startsWith("/")) {
+                                throw new RuntimeException(
+                                        "Resources cannot start with /, " + i + " is incorrect provided by " + element);
+                            }
+                            List<ClassPathElement> list = elementMap.get(i);
+                            if (list == null) {
+                                elementMap.put(i, list = new ArrayList<>());
+                            }
+                            list.add(element);
+                        }
+                    }
+                    Map<String, ClassPathElement[]> finalElements = new HashMap<>();
+                    for (Map.Entry<String, List<ClassPathElement>> i : elementMap.entrySet()) {
+                        finalElements.put(i.getKey(), i.getValue().toArray(new ClassPathElement[i.getValue().size()]));
+                    }
+                    Set<String> banned = new HashSet<>();
+                    for (ClassPathElement i : bannedElements) {
+                        banned.addAll(i.getProvidedResources());
+                    }
+                    Set<String> parentFirstResources = new HashSet<>();
+                    for (ClassPathElement i : parentFirstElements) {
+                        parentFirstResources.addAll(i.getProvidedResources());
+                    }
+                    return this.state = new ClassLoaderState(finalElements, banned, parentFirstResources);
+                }
+            }
+        }
+        return state;
+    }
+
+    @Override
+    public URL getResource(String nm) {
+        String name = sanitizeName(nm);
+        ClassLoaderState state = getState();
+        if (state.bannedResources.contains(name)) {
+            return null;
+        }
+        //        ClassPathElement[] providers = loadableResources.get(name);
+        //        if (providers != null) {
+        //            return providers[0].getResource(nm).getUrl();
+        //        }
+        //TODO: because of dev mode we can't use the fast path her, we need to iterate
+        for (ClassPathElement i : elements) {
+            ClassPathResource res = i.getResource(name);
+            if (res != null) {
+                return res.getUrl();
+            }
+        }
+        return parent.getResource(nm);
```

Comment:
How about taking the value of `aggregateParentResources` into account here and in `getResourceAsStream` below? Wondering whether it's this way on purpose. Otherwise, it'd be probably a good idea to have a consistent resource access switch.

Reply:
So we can't completely hide the JDK resources or ArC can't index them, but we want to hide duplicate/unwanted copies of META-INF/services files. I was not 100% sure about this but I can't really think of a better approach.

Reply:
Ok